WordPress/Classes/Utility/AppIcon.swift

```diff
@@ -13,6 +13,9 @@ struct AppIcon {
     /// by a newer style of icon, but are still provided for compatibility and for users who prefer them.
     let isLegacy: Bool
 
+    // Boolean indicating whether the icon is the default icon of the app.
+    let isPrimary: Bool
```

Comment:
👋 Hey @salimbraksa - is it a true statement given the array of all possible icons in the app, only one will have `isPrimary` set to `true`?

If so, I wonder if this should be more constrained so somehow in the future multiple don't have their value set to `true`.

Reply:
@twstokes Yeah I agree this approach is a bit prone to bugs, because it's programmatically possible to have 2 icons marked as `primary`. 

## Other Approaches Considered
Actually the first approach I tried is more constrained but it had some drawbacks:

Before adding the `isPrimary` property, I checked whether an icon is primary or not by comparing an icon instance with the static default icon instance, like this:

```swift
let isPrimary = someIconInstance.name == AppIcon.defaultIcon.name
```
That code was executed every time an icon is selected, and as you know comparing strings is definitely slower than just comparing booleans.

Also, It didn't make sense to me compute `isPrimary` every time we want to use, an icon is either primary or not and it won't change at runtime. We could solve this problem by having a lazy `isPrimary` property, like this:

```swift
class AppIcon {
      private(set) lazy var isPrimary: Bool = self.name == AppIcon.defaultIcon.name
}
```
It's still not 100% constrained and it still relies on the string comparison, but at least `isPrimary` is no longer a computed property.

## Preferred Approach
Since we want to make sure only 1 primary icon can exist, I added a few unit tests to cover that case https://github.com/wordpress-mobile/WordPress-iOS/pull/19378/commits/ccc7534ae030044affdc42cecd5551b740441e5c and I kept `isPrimary` as an instance property because it performs better than the other approaches.

Let me know what you think and I'm happy to make changes if you prefer another approach 🙏

Reply:
Thanks @salimbraksa for your thoughts! Adding the tests is a good call, and this isn't a blocker so it all LGTM. 🙇